### PR TITLE
CI housekeeping

### PR DESCRIPTION
**Pull request by kianmeng**
List of changes:
- add Elixir 1.17 / Erlang OTP 27 to CI
- bump GitHub actions

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -9,7 +9,7 @@ defmodule BroadwayCloudPubSub.MixProject do
     [
       app: :broadway_cloud_pub_sub,
       version: @version,
-      elixir: "~> 1.11",
+      elixir: "~> 1.12",
```
Comment (kianmeng) [inline mix.exs:12]: Bump to 1.12 where the security support just ended recently, see https://endoflife.date/elixir

Not sure if we should sync with Broadway min Elixir requirement, which is still set to 1.7, see https://github.com/dashbitco/broadway/blob/9eaf2140b6c1f36b25f990a7c74052bece26aadc/mix.exs#L11

Comment (josevalim) [reply mix.exs:12]: It is end of life but we should not be the ones forcing users to update. If we can still support 1.11, then it should. :)

Comment (kianmeng) [reply mix.exs:12]: Noted, and updated.

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Comment by kianmeng** [comment]
:partying_face: :partying_face: :partying_face: :partying_face: :partying_face:

**Pull request merged**